common/src/main/java/org/apache/celeborn/common/network/server/TransportRequestHandler.java

```diff
@@ -73,8 +76,64 @@ public void channelInactive() {
 
   @Override
   public void handle(RequestMessage request) {
+    logger.trace("Received request {} from {}", request.getClass().getName(), reverseClient);
     if (checkRegistered(request)) {
-      msgHandler.receive(reverseClient, request);
+      if (request instanceof RpcRequest) {
+        processRpcRequest((RpcRequest) request);
+      } else if (request instanceof OneWayMessage) {
+        processOneWayMessage((OneWayMessage) request);
+      } else {
+        processOtherMessages(request);
+      }
+    }
+  }
+
+  private void processRpcRequest(final RpcRequest req) {
```

Comment:
what's the rationale to declare final for `req` in `processRpcRequest` but not for the same in other two processing methods(`processOneWayMessage` and `processOtherMessages`)?

Reply:
req is being used in an anonymous callback and variables used within anonymous inner classes must be final or effectively final so this is why this is declared as final here.